service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -190,8 +207,143 @@ public void prepareCustomDirections(Map<String, Object> data) {
+            judgeAndLegalAdvisor), EMPTY_PLACEHOLDER));
+
+        cmoTemplateData.putAll(getGroupedCMODirections(caseData));
+
+        cmoTemplateData.put("draftbackground", String.format("image:base64:%1$s",
+            docmosisDocumentGeneratorService.generateDraftWatermarkEncodedString()));
+
+        List<Map<String, String>> recitals = buildRecitals(caseManagementOrder.getRecitals());
+        cmoTemplateData.put("recitals", recitals);
+        cmoTemplateData.put("recitalsProvided", recitals.size());
+
+        cmoTemplateData.put("schedule", caseManagementOrder.getSchedule());
+        cmoTemplateData.put("scheduleProvided", isNotEmpty(caseManagementOrder.getSchedule()));
+
+        //defaulting as 1 as we currently do not have impl for multiple CMos
+        cmoTemplateData.put("caseManagementNumber", 1);
+
+        return cmoTemplateData.build();
+    }
+
+    private JudgeAndLegalAdvisor getJudgeAndLegalAdvisor(final CaseData caseData,
+                                                         final CaseManagementOrder caseManagementOrder) {
+        return isNotEmpty(caseManagementOrder)
+            ? defaultIfNull(caseManagementOrder.getJudgeAndLegalAdvisor(), null) :
+            caseData.getJudgeAndLegalAdvisor();
+    }
+
+    private Map<String, List<Map<String, String>>> getGroupedCMODirections(CaseData caseData) throws IOException {
+        OrderDefinition caseManagementOrderDefinition = ordersLookupService.getDirectionOrder();
+
+        if (isNull(caseData.getCaseManagementOrder())) {
+            return ImmutableMap.of();
+        }
+
+        CaseManagementOrder caseManagementOrder = caseData.getCaseManagementOrder();
+
+        Map<DirectionAssignee, List<Element<Direction>>> groupedDirections =
+            directionHelperService.sortDirectionsByAssignee(directionHelperService.numberDirections(
+                caseManagementOrder.getDirections()));
+
+        ImmutableMap.Builder<String, List<Map<String, String>>> formattedDirections = ImmutableMap.builder();
+
+        groupedDirections.forEach((key, value) -> {
+            List<Map<String, String>> directionsList = value.stream()
+                .map(Element::getValue)
+                .filter(direction -> !"No".equals(direction.getDirectionNeeded()))
+                .map(direction -> ImmutableMap.of(
+                    "title", caseDataExtractionService.formatTitle(
+                        direction, caseManagementOrderDefinition.getDirections()),
+                    "body", defaultIfNull(direction.getDirectionText(), EMPTY_PLACEHOLDER)))
+                .collect(toList());
+
+            formattedDirections.put(key.getValue(), directionsList);
+        });
+
+        return formattedDirections.build();
+    }
+
+    private String getFirstRespondentFullName(final CaseData caseData) {
+        return caseData.buildFirstRespondentFullName();
+    }
+
+    private Map<String, Object> getHearingBooking(final CaseData caseData, DynamicList hearingDateList) {
```

Comment:
Be consistent, either you declare args as final or not.

Reply:
final for everything 😄